src/frontend/src/views/Upload/components/Metadata/components/StaticTable/index.tsx

```diff
@@ -50,78 +62,132 @@ const validationSchema = object({
 export default function StaticTable({
   metadata,
   setIsValid,
+  hasImportedMetadataFile,
 }: Props): JSX.Element {
+  const [validationErrors, setValidationErrors] =
+    useState<Record<string, ValidationErrorRecord | null>>(EMPTY_OBJECT);
+
+  // This function validates metadata by creating a mapping of sample ids to
+  // either null or a ValidationErrorRecord, an object which is itself a mapping
+  // from the key(s) that failed to validate to the specific error message.
+  // The function then determines if the metadata as a whole is valid by
+  // checking that each sampleId maps to null.
+  // If one field has multiple error messages we just overwrite the message in
+  // the ValidationErrorRecord, since we only want to display one error at a time.
   const validateMetadata = useCallback(
     async (metadata: SampleIdToMetadata | null) => {
       if (metadata == null) {
         setIsValid(false);
         return;
       }
-      const rowValidation: Record<string, boolean> = {};
+      const validationErrors: Record<string, ValidationErrorRecord | null> =
```

Comment:
I'm a bit torn on this process. Looking over it, everything looks good, and it does seem nice to be able to use Yup for the validation since that matches how the dynamic table works. That said, because it's a static table, we can rely on the user passing data through the metadata `parseFile` function, which already does validation. But on the other hand, I know moving those warnings around is a pain and might introduce extra complexity if it's doing extra checks for `>100` samples, so what's here might be the best way. But if you've already looked into that and this seemed best, then I'm happy.

Reply:
The user does pass data through the metadata parseFile function. It probably would have been better overall to validate there and surface any errors, but it also would have been much more complex. I went for the simplest route.